src/CDataInterface/format_string_to_types.jl

```diff
@@ -0,0 +1,68 @@
+# https://arrow.apache.org/docs/format/CDataInterface.html#data-type-description-format-strings
+module FormatStrings
+
+function get_type_from_format_string(format_string ::String) ::DataType
+    if format_string == "n"
+        Nothing
+    elseif format_string == "b"
+        Bool
+    elseif format_string == "c"
+        Int8
+    elseif format_string == "C"
+        UInt8
+    elseif format_string == "s"
+        Int16
+    elseif format_string == "S"
+        UInt16
+    elseif format_string == "i"
+        Int32
+    elseif format_string == "I"
+        UInt32
+    elseif format_string == "l"
+        Int64
+    elseif format_string == "L"
+        UInt64
+    elseif format_string == "e"
+        Float16
+    elseif format_string == "f"
+        Float32
+    elseif format_string == "g"
+        Float64
+    elseif format_string == "z" || format_string == "Z"
+        Vector{UInt8}
+    elseif format_string == "u" || format_string == "U"
+        String
+    elseif format_string[1] == 'd'
+        const splits = split(format_string[3:end], ",")
+        precision = Int(splits[1])
+        scale = Int(splits[2])
+        if length(splits) == 3
+            bandwidth = splits[3]
+        end
+        #TODO return something here
+    elseif format_string[1] == 'w'
+        #TODO figure out fixed width binary
+    elseif format_string[1] == '+'
+        if format_string[2] == 'l' || format_string[2] == 'L'
+            Arrow.List
```

Comment:
For these nested types, we'll need to parse the children recursively. So the overall method signature needs to take the full `ArrowSchema` type, and then access the format string at the top. Then when we get here, we'll call `get_type_from_format_string(sch.children[1])` and so on to get the List element type so we end up with a type like `Arrow.List{Vector{Int64}}` or whatever.